### ISSUE TITLE
What's the proper way to download a file in a razzle environment?

### ISSUE THREAD

**Issue by strommj** [jaredpalmer/razzle#1220]
After much turmoil on this otherwise seemingly simple issue, my application still cannot download a file when it's pushed to my hosting server and in production.  I have reason to believe at this point that it's something to do with Razzle and/or SSR, as everything on the internet about downloading a file from the public directory points to using a simple anchor tag pointing to the file location.  And since Razzle doesn't have any presence on stackoverflow, I'll post my question here.

What happens is essentially Instead of downloading the file from the public folder, it downloads some arbitrary saved (offline) html page.  The issue is present on the hosted server only, local machine is fine (desktop dpes seem to download fine, but I haven't tested the downloaded apk yet.  Mobile is the main issue, and thus I'm assuming the link downloaded on the desktop is also incorrect).  

I have eventually resorted to this method:

```javascript
<Button href={process.env.PUBLIC_PATH + "app.apk"} download="myDownload.apk">Download</Buton>
```

I've also tried many variations of the path: `build/public/app.apk`, `/build/public/app.apk`, `public/app.apk`, `app.apk`, `/app.apk`, as well as other `env` variables such as `PUBLIC_URL`, `RAZZLE_PUBLIC_DIR`, and `CLIENT_PUBLIC_PATH`.  (Some of these obviously don't work, some work on local machine only.)

When inspecting the button (on the hosted environment) it shows `<a href="build/public/app.apk" ... />` instead of `<a href="/app.apk" ... />`.

If anybody wants to see my website to see the above behavior 

The hosted environment file system looks something like this:

- file system root/
    - build/
        - public/
            - static/ <-- client bundle code lives here
            - app.apk
        - server.js <-- server runs app from here

If anyone wants to have a look at the issue themselves, take a look [here](https://www.musicians-aide.app).  (For privacy, I'll remove this link after some time).  Download link is at the bottom of the page.

**Comment by nimaa77** [comment]
@strommj hello,

just do this:

project/
  - public/
    - app.apk <- your apk file 
  - src/
  - build/

and then inside your component just do this:
```jsx
<Button href="/app.apk" download="myDownload.apk">Download</Buton>
```

Just close the issue if this worked for you

GG 😎

**Comment by strommj** [comment]
Thanks nimaa77 for the reply.  If you look to my question, I have already tried that.  Again, I can only download an app file on my local instance of my website, but from my production server, I cannot download the file correctly on mobile, see my above question.

**Comment by nimaa77** [comment]
could you please create an example with [https://codesandbox.io/s/github/jaredpalmer/razzle/tree/master/examples/basic](https://codesandbox.io/s/github/jaredpalmer/razzle/tree/master/examples/basic ) this and send a link here

**Comment by nimaa77** [comment]
@strommj but are you sure about the folder you put the apk file into it?
in your question, it's in /build folder but I said put it inside the /public folder (not in /build/public)

**Comment by strommj** [comment]
I can try to see if I can make a simple example tonight.

As to your second point: I do have it in my public folder.  When the code is pushed to the server, and `npm run build` is run, its final resting place thus is within the build directory, as this is how the website is run on my hosting server (see my above question).

Either way, when the website is on the hosting server being ran under the build directory, it should be the same as what you say, and the PUBLIC_PATH environment variable should be able to correctly point to the right place.

**Comment by strommj** [comment]
@nimaa77 I am unable to create an example, because I cannot run a minimized production build on that sandbox.

**Comment by nimaa77** [comment]
just create an example project and put it inside a Github repo and send me the link

**Comment by strommj** [comment]
Ok sorry for the delay on this.  I created a simplified version of my website, and thus far I haven't been able to recreate this issue.

https://github.com/strommj/razzle-basic-nav

I will continue to incorporate more aspects of my website and can re-open this issue if I find more information about the problem.

I have my suspicions that it was due to a second root level server.js file I had that seemed to be handling some routes to my website.  I'm not actually sure where I got that piece of code, but your basic example doesn't have anything like that, so I'm hoping that was the culprit.

I can also try to consolidate this information I went through and create another example for the razzle repository here at least for the navigation setup I have
